### ISSUE TITLE
[BUG]: Regression: Driv3r (both USA and PAL versions) crashes the emulator when in hardware mode

### ISSUE THREAD

**Issue by Shideravan** [PCSX2/pcsx2#6037]
### Describe the Bug

This game alongside Stuntman are know to not running well with hardware mode. The novelty seems that now the games crashes/freezes the emulator completely in the builds.

### Reproduction Steps

1. Initiate the game using hardware mode for graphics (Vulkan, DirectX12, etc)
2. Go until gameplay section. Before the gameplay section starts, the emulator will freeze/crash
3. It's valid to note that the VRAM will be completely used. Probably it's a memory leak

### Expected Behavior

The emulator at least don't crash when using the hardware mode in Driv3r

### PCSX2 Revision

v.1.7.2721

### Operating System

Windows 10 (64bit)

### CPU

i7-4790

### GPU

GTX 1070

### GS Settings

![image](https://user-images.githubusercontent.com/12163923/167304137-8dfcb648-ecbe-4b84-bdde-ff59e44edd20.png)
![image](https://user-images.githubusercontent.com/12163923/167304149-71289f69-aefa-44e9-ae5a-9ad3d22e9eba.png)
No hacks, OSD, recording or advanced setting changed

### Emulation Settings

Everything on default

### Logs & Dumps

Not possible to GSdump due emulator freezing/crashing
emuLog indicates GS: Memory allocation failure
[emuLog.txt](https://github.com/PCSX2/pcsx2/files/8647315/emuLog.txt)
.

**Comment by refractionpcsx2** [comment]
The game does seem to require a lot of VRAM, mine uses about 4GB at 4x, I would certainly recommend lowering your resolution a little.  That much usage might be correct depending on what the game does with the textures.

**Comment by Shideravan** [comment]
Thanks for the feedback. Yes, reducing the upscale a little helps with crashes of the emulator. But this particular overuse of VRAM seems unnatural, don't is? My videoboard (GTX 1070) has 8GB VRAM and it's completely full when entering ingame in 4X scale to the point of crashes the emulator...

**Comment by Mrlinkwii** [comment]
@Shideravan  apparently the game has  texture cache issues , id recommend playing the game in SW mode  for the time being

**Comment by Nerboruto** [comment]
it happened to me on windows 10 too.
some games give this error when playing a cutscene es.. metal slug 3 or Ace Combat: The Belkan War
Ace Combat: Squadron Leader works perfectly without this error.
I have tried various backends graphics api and both the qt and vx version.
on qt a few times I managed to get out of the error by pressing esc and switching to the overlay and returning to the game.
on wx I think it can not be done.
resolution is irrelevant it also does the native one.
[emuLog.txt](https://github.com/PCSX2/pcsx2/files/9565003/emuLog.txt)
